atomic_reactor/utils/operator.py

```diff
@@ -115,32 +269,40 @@ def name(self):
     def description(self):
         return "{} annotation".format(self._image_key)
 
-    def with_key(self, image_key):
+    def in_key(self, image_key, start=None, end=None):
         self._image_key = image_key
+        self._start_i = start or 0
+        self._end_i = end or len(self.data[image_key])
         return self
 
 
 class OperatorCSV(object):
     """
     A single ClusterServiceVersion file in an operator manifest.
 
-    Can find and replace pullspecs for container images in predefined locations.
+    Can find and replace pullspecs for container images in predefined locations
+    and all annotations (using a heuristic to guess what is a pullspec).
     """
 
-    def __init__(self, path, data):
+    def __init__(self, path, data, pullspec_heuristic=None):
         """
         Initialize an OperatorCSV
 
         :param path: Path where data was found or where it should be written
         :param data: ClusterServiceVersion yaml data
+        :param pullspec_heuristic: Function that takes a string and returns
+            a list of pullspecs (substrings). If not specified, a default
+            implementation will be used. For more information, see the
+            default_pullspec_heuristic() function in this module.
         """
         if data.get("kind") != OPERATOR_CSV_KIND:
             raise NotOperatorCSV("Not a ClusterServiceVersion")
         self.path = path
         self.data = data
+        self._pullspec_heuristic = pullspec_heuristic or default_pullspec_heuristic
 
     @classmethod
-    def from_file(cls, path):
+    def from_file(cls, path, **kwds):
```

Comment:
`**kwargs` is common term?

Reply:
Same for `**kwds`: https://docs.python.org/3/library/functools.html#functools.wraps
But I don't really have a preference